2025-08-17 15:49:35,066 - INFO - GET ALL - Listagem de Salas
2025-08-17 15:49:35,089 - ERROR - Erro SQLAlchemy ao buscar Salas:
	Traceback (most recent call last):
	  File "/home/davi/Downloads/keyControlApi/venv/lib/python3.12/site-packages/sqlalchemy/orm/clsregistry.py", line 516, in _resolve_name
	    rval = d[token]
	           ~^^^^^^^
	  File "/home/davi/Downloads/keyControlApi/venv/lib/python3.12/site-packages/sqlalchemy/util/_collections.py", line 345, in __missing__
	    self[key] = val = self.creator(key)
	                      ^^^^^^^^^^^^^^^^^
	  File "/home/davi/Downloads/keyControlApi/venv/lib/python3.12/site-packages/sqlalchemy/orm/clsregistry.py", line 484, in _access_cls
	    return self.fallback[key]
	           ~~~~~~~~~~~~~^^^^^
	KeyError: 'Professor'
	
	The above exception was the direct cause of the following exception:
	
	Traceback (most recent call last):
	  File "/home/davi/Downloads/keyControlApi/resources/SalaResource.py", line 16, in get
	    salas = db.session.execute(query).scalars().all()
	            ^^^^^^^^^^^^^^^^^^^^^^^^^
	  File "/home/davi/Downloads/keyControlApi/venv/lib/python3.12/site-packages/sqlalchemy/orm/scoping.py", line 779, in execute
	    return self._proxied.execute(
	           ^^^^^^^^^^^^^^^^^^^^^^
	  File "/home/davi/Downloads/keyControlApi/venv/lib/python3.12/site-packages/sqlalchemy/orm/session.py", line 2365, in execute
	    return self._execute_internal(
	           ^^^^^^^^^^^^^^^^^^^^^^^
	  File "/home/davi/Downloads/keyControlApi/venv/lib/python3.12/site-packages/sqlalchemy/orm/session.py", line 2251, in _execute_internal
	    result: Result[Any] = compile_state_cls.orm_execute_statement(
	                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
	  File "/home/davi/Downloads/keyControlApi/venv/lib/python3.12/site-packages/sqlalchemy/orm/context.py", line 306, in orm_execute_statement
	    result = conn.execute(
	             ^^^^^^^^^^^^^
	  File "/home/davi/Downloads/keyControlApi/venv/lib/python3.12/site-packages/sqlalchemy/engine/base.py", line 1415, in execute
	    return meth(
	           ^^^^^
	  File "/home/davi/Downloads/keyControlApi/venv/lib/python3.12/site-packages/sqlalchemy/sql/elements.py", line 523, in _execute_on_connection
	    return connection._execute_clauseelement(
	           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
	  File "/home/davi/Downloads/keyControlApi/venv/lib/python3.12/site-packages/sqlalchemy/engine/base.py", line 1629, in _execute_clauseelement
	    compiled_sql, extracted_params, cache_hit = elem._compile_w_cache(
	                                                ^^^^^^^^^^^^^^^^^^^^^^
	  File "/home/davi/Downloads/keyControlApi/venv/lib/python3.12/site-packages/sqlalchemy/sql/elements.py", line 711, in _compile_w_cache
	    compiled_sql = self._compiler(
	                   ^^^^^^^^^^^^^^^
	  File "/home/davi/Downloads/keyControlApi/venv/lib/python3.12/site-packages/sqlalchemy/sql/elements.py", line 320, in _compiler
	    return dialect.statement_compiler(dialect, self, **kw)
	           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
	  File "/home/davi/Downloads/keyControlApi/venv/lib/python3.12/site-packages/sqlalchemy/sql/compiler.py", line 1446, in __init__
	    Compiled.__init__(self, dialect, statement, **kwargs)
	  File "/home/davi/Downloads/keyControlApi/venv/lib/python3.12/site-packages/sqlalchemy/sql/compiler.py", line 886, in __init__
	    self.string = self.process(self.statement, **compile_kwargs)
	                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
	  File "/home/davi/Downloads/keyControlApi/venv/lib/python3.12/site-packages/sqlalchemy/sql/compiler.py", line 932, in process
	    return obj._compiler_dispatch(self, **kwargs)
	           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
	  File "/home/davi/Downloads/keyControlApi/venv/lib/python3.12/site-packages/sqlalchemy/sql/visitors.py", line 141, in _compiler_dispatch
	    return meth(self, **kw)  # type: ignore  # noqa: E501
	           ^^^^^^^^^^^^^^^^
	  File "/home/davi/Downloads/keyControlApi/venv/lib/python3.12/site-packages/sqlalchemy/sql/compiler.py", line 4728, in visit_select
	    compile_state = select_stmt._compile_state_factory(
	                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
	  File "/home/davi/Downloads/keyControlApi/venv/lib/python3.12/site-packages/sqlalchemy/sql/base.py", line 687, in create_for_statement
	    return klass.create_for_statement(statement, compiler, **kw)
	           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
	  File "/home/davi/Downloads/keyControlApi/venv/lib/python3.12/site-packages/sqlalchemy/orm/context.py", line 447, in create_for_statement
	    return cls._create_orm_context(
	           ^^^^^^^^^^^^^^^^^^^^^^^^
	  File "/home/davi/Downloads/keyControlApi/venv/lib/python3.12/site-packages/sqlalchemy/orm/context.py", line 1175, in _create_orm_context
	    _QueryEntity.to_compile_state(
	  File "/home/davi/Downloads/keyControlApi/venv/lib/python3.12/site-packages/sqlalchemy/orm/context.py", line 2628, in to_compile_state
	    _MapperEntity(
	  File "/home/davi/Downloads/keyControlApi/venv/lib/python3.12/site-packages/sqlalchemy/orm/context.py", line 2708, in __init__
	    entity._post_inspect
	  File "/home/davi/Downloads/keyControlApi/venv/lib/python3.12/site-packages/sqlalchemy/util/langhelpers.py", line 1338, in __get__
	    obj.__dict__[self.__name__] = result = self.fget(obj)
	                                           ^^^^^^^^^^^^^^
	  File "/home/davi/Downloads/keyControlApi/venv/lib/python3.12/site-packages/sqlalchemy/orm/mapper.py", line 2724, in _post_inspect
	    self._check_configure()
	  File "/home/davi/Downloads/keyControlApi/venv/lib/python3.12/site-packages/sqlalchemy/orm/mapper.py", line 2401, in _check_configure
	    _configure_registries({self.registry}, cascade=True)
	  File "/home/davi/Downloads/keyControlApi/venv/lib/python3.12/site-packages/sqlalchemy/orm/mapper.py", line 4214, in _configure_registries
	    _do_configure_registries(registries, cascade)
	  File "/home/davi/Downloads/keyControlApi/venv/lib/python3.12/site-packages/sqlalchemy/orm/mapper.py", line 4255, in _do_configure_registries
	    mapper._post_configure_properties()
	  File "/home/davi/Downloads/keyControlApi/venv/lib/python3.12/site-packages/sqlalchemy/orm/mapper.py", line 2418, in _post_configure_properties
	    prop.init()
	  File "/home/davi/Downloads/keyControlApi/venv/lib/python3.12/site-packages/sqlalchemy/orm/interfaces.py", line 589, in init
	    self.do_init()
	  File "/home/davi/Downloads/keyControlApi/venv/lib/python3.12/site-packages/sqlalchemy/orm/relationships.py", line 1656, in do_init
	    self._setup_entity()
	  File "/home/davi/Downloads/keyControlApi/venv/lib/python3.12/site-packages/sqlalchemy/orm/relationships.py", line 1866, in _setup_entity
	    self._clsregistry_resolve_name(argument)(),
	    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
	  File "/home/davi/Downloads/keyControlApi/venv/lib/python3.12/site-packages/sqlalchemy/orm/clsregistry.py", line 520, in _resolve_name
	    self._raise_for_name(name, err)
	  File "/home/davi/Downloads/keyControlApi/venv/lib/python3.12/site-packages/sqlalchemy/orm/clsregistry.py", line 501, in _raise_for_name
	    raise exc.InvalidRequestError(
	sqlalchemy.exc.InvalidRequestError: When initializing mapper Mapper[Reserva(reserva)], expression 'Professor' failed to locate a name ('Professor'). If this is a class name, consider adding this relationship() to the <class 'models.Reserva.Reserva'> class after both dependent classes have been defined.
	
2025-08-17 15:52:54,657 - INFO - GET ALL - Listagem de Salas
2025-08-17 15:52:54,680 - INFO - Salas retornadas com sucesso
2025-08-17 15:53:30,776 - INFO - GET ALL - Listagem de Salas
2025-08-17 15:53:30,833 - INFO - Salas retornadas com sucesso
2025-08-17 15:53:33,834 - INFO - GET BY sala_id - Sala (1)
2025-08-17 15:53:37,153 - INFO - GET ALL - Listagem de Salas
2025-08-17 15:53:37,157 - INFO - Salas retornadas com sucesso
2025-08-17 15:53:42,611 - INFO - GET ALL - Listagem de Responsaveis
2025-08-17 15:53:42,614 - INFO - Responsaveis retornadas com sucesso
2025-08-17 15:53:46,383 - INFO - GET BY responsavel_id - Responsavel 1
2025-08-17 15:54:04,834 - INFO - GET ALL - Listagem de Reservas
2025-08-17 15:54:04,838 - INFO - Reservas retornadas com sucesso
2025-08-17 15:54:08,460 - INFO - GET BY reserva_id - Reserva 1
2025-08-17 15:55:12,723 - INFO - GET ALL - Listagem de Responsaveis
2025-08-17 15:55:12,726 - INFO - Responsaveis retornadas com sucesso
2025-08-17 15:55:41,998 - INFO - POST - Novo Responsavel
2025-08-17 15:55:53,232 - INFO - GET ALL - Listagem de Responsaveis
2025-08-17 15:55:53,233 - INFO - Responsaveis retornadas com sucesso
2025-08-17 15:55:56,124 - INFO - GET BY responsavel_id - Responsavel 1
2025-08-17 15:58:15,820 - INFO - GET ALL - Listagem de Salas
2025-08-17 15:58:15,822 - INFO - Salas retornadas com sucesso
2025-08-17 16:00:04,458 - INFO - POST - Nova Sala
2025-08-17 16:00:04,468 - INFO - Sala 1 criada com sucesso!
2025-08-17 16:02:16,305 - INFO - POST - Nova reserva
2025-08-17 16:02:33,431 - INFO - POST - Nova reserva
2025-08-17 16:09:07,454 - INFO - POST - Nova reserva
2025-08-17 16:09:25,399 - INFO - POST - Nova reserva
